Sources/JWTKit/JWTSigners.swift

```diff
@@ -108,8 +109,99 @@ public final class JWTSigners {
     {
         try JWTParser(token: token).payload(as: Payload.self)
     }
-
-
+    
+    public func verifyJWSWithX5C<Payload>(
+        _ token: String,
+        as payload: Payload.Type = Payload.self,
+        rootCert: String
+    ) throws -> Payload
+        where Payload: JWTPayload
+    {
+        try self.verifyJWSWithX5C([UInt8](token.utf8), as: Payload.self, rootCert: [UInt8](rootCert.utf8))
+    }
+    
+    func addBoundaryToCert(_ cert: String) -> String {
+        """
+        -----BEGIN CERTIFICATE-----
+        \(cert)
+        -----END CERTIFICATE-----
+        """
+    }
+    
+    public func verifyJWSWithX5C<Message, Payload>(
+        _ token: Message,
+        as payload: Payload.Type = Payload.self,
+        rootCert: Message
+    ) throws -> Payload
+        where Message: DataProtocol, Payload: JWTPayload
+    {
+        let parser = try JWTParser(token: token)
+        let header = try parser.header()
+        guard let x5c = header.x5c else {
+            throw JWTError.generic(identifier: "JWS", reason: "No x5c certificates provided")
+        }
+        
+        // Verify the chain
+        // The first cert is used to sign the JWS
+        // Each subsequent cert should be used to certify the previous one
+        // For the last cert we use the root cert passed in
+        // https://datatracker.ietf.org/doc/html/rfc7515#section-4.1.6
+        for (index, certificate) in x5c.enumerated() {
+            if index == 0 {
+                continue
+            }
+            
+            if index == x5c.count {
```

Comment:
```suggestion
            if index == x5c.count - 1 {
```
Otherwise this code won't be run, since `x5c.enumerated().last?.offset == x5c.count - 1`. This means that the root cert is never checked. So any valid chain is verified, even if it's not using the trusted root cert.